emscripten.py

```diff
@@ -586,6 +605,9 @@ def create_em_js(metadata):
     arg_names = [arg.split()[-1].replace("*", "") for arg in args if arg]
     args = ','.join(arg_names)
     func = f'function {name}({args}) {body}'
+    if settings.ASYNCIFY == 2 and name in metadata['emJsFuncsTypeId']:
+      typeId = metadata['emJsFuncsTypeId'][name]
+      func = func + f'\n{name}.type = EM_JS_FUNC_TYPES[{typeId}];'
```

Comment:
Why not just use the existing `.sig = <string` mechanism that we already have?   I seem more compact and we already have `sigToWasmTypes` converting to this format at runtime.

Reply:
Oh yeah I wanted to ask you about this. With the metadata extraction we only get the wasm type so we lose information about pointers, so the `sig` won't really match one of the hardcoded ones we have. It seems like that could lead to some confusion down the road. Fine with me though if you aren't concerned.

Reply:
@sbc100 what are you thoughts here?

Reply:
I think having `sig` without the pointer information is fine in this case.  At least no worse that what we have today for EM_JS functions which is not signature at all.  

EM_JS function cannot have explicit user-specified sig attributes since that only works for functions defined in library files (I think), so you don't need to worry about conflicts with a user-specified sig.